components-starter/camel-hashicorp-vault-starter/src/main/java/org/apache/camel/component/hashicorp/vault/springboot/SpringBootHashicorpVaultPropertiesParser.java

```
@@ -0,0 +1,80 @@
+
+            VaultEndpoint vaultEndpoint = new VaultEndpoint();
+            vaultEndpoint.setHost(host);
+            vaultEndpoint.setPort(port);
+            vaultEndpoint.setScheme(scheme);
+
+            VaultTemplate client = new VaultTemplate(
+                    vaultEndpoint,
+                    new TokenAuthentication(token));
+            HashicorpVaultPropertiesFunction hashicorpVaultPropertiesFunction = new HashicorpVaultPropertiesFunction(client);
+
+            final Properties props = new Properties();
+            for (PropertySource mutablePropertySources : event.getEnvironment().getPropertySources()) {
+                if (mutablePropertySources instanceof MapPropertySource mapPropertySource) {
+                    mapPropertySource.getSource().forEach((key, value) -> {
+                        String stringValue = null;
+                        if ((value instanceof OriginTrackedValue originTrackedValue &&
+                                originTrackedValue.getValue() instanceof String v)) {
+                            stringValue = v;
+                        } else if (value instanceof String v) {
+                            stringValue = v;
+                        }
+
+                        if (stringValue != null &&
+                                stringValue.startsWith("{{hashicorp:") &&
+                                stringValue.endsWith("}}")) {
+                            LOG.debug("decrypting and overriding property {}", key);
+                            try {
+                                props.put(key, hashicorpVaultPropertiesFunction.apply(stringValue
+                                        .replace("{{hashicorp:", "")
+                                        .replace("}}", "")));
+                            } catch (Exception e) {
+                                // Log and do nothing
+                                LOG.debug("failed to parse property {}", key, e);
```

Add ". This exception is ignored".